app/presenters/tree_builder.rb

```diff
@@ -480,7 +480,10 @@ def self.rbac_filtered_objects(objects, options = {})
 
     # Remove VmOrTemplate :match_via_descendants option if present, comment to let Rbac.search process it
     descendants = false
-    descendants = options.delete(:match_via_descendants) if %w(ConfiguredSystems VmOrTemplate).include?(options[:match_via_descendants])
+    descendant = options[:match_via_descendants]
+    descendant = descendant.first if descendant.kind_of?(Array)
+    descendants = options.delete(:match_via_descendants) if %w(ConfiguredSystem ConfiguredSystems
+                                                               VmOrTemplate).include?(descendant.to_s)
```

Comment:
@kbrock It looks like we do not need this whole ```descendant``` logic at all.
The above method should really be cleaned down to -
```ruby
def self.rbac_filtered_objects(objects, options = {})
    return objects if objects.empty?
    results = Rbac.filtered(objects, options)
    results
end
```